src/Adapter/MSTest.TestAdapter/Execution/TestMethodRunner.cs

```diff
@@ -72,7 +75,7 @@ public TestMethodRunner(TestMethodInfo testMethodInfo, TestMethod testMethod, IT
     /// Executes a test.
     /// </summary>
     /// <returns>The test results.</returns>
-    internal UnitTestResult[] Execute()
+    internal UnitTestResult[] Execute(ClassCleanupManager? _classCleanupManager, TypeCache _typeCache)
```

Comment:
This method is getting super complex. I would suggest that you extract the content into sub-methods.